### ISSUE TITLE
Issues with fact_path: Parameter 'fact_path' has an invalid path..

### ISSUE THREAD

**Issue by erik-saarts** [ansible/ansible#37712]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

setup

##### ANSIBLE VERSION

```
ansible 2.6.0 (devel dd0140dfed) last updated 2018/03/14 16:53:57 (GMT +300)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/tmp/ansible/lib/ansible
  executable location = /root/tmp/ansible/bin/ansible
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]

```

##### CONFIGURATION

DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = [u'/root/windows/hosts']

##### OS / ENVIRONMENT

Ansible running in Debian 9.3
Remote hosts: windows 10 professional version 1709 

##### SUMMARY

When using setup module to gather facts from Windows 10 Pro remote hosts, giving 'facts_path' variable produces confusing errors in certain cases. 

##### STEPS TO REPRODUCE

```yaml
ansible mygroup -m setup -a "fact_path='C:\a'"
```

##### EXPECTED RESULTS

Output of ansible facts, along with the data from the scripts residing in C:\a or error message about
non-existing path on remote machine. 
##### ACTUAL RESULTS

Certain combinations of characters in fact_path, like first symbol after backslash being either 'a','b','f','n','r','t' or 'v', seem to be not tolerated. Therefore, 'C:\ca' works correctly but 'C:\ac' does not.
Capital letters can be used as a workaround.

```
ansible 2.6.0 (devel dd0140dfed) last updated 2018/03/14 16:53:57 (GMT +300)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/tmp/ansible/lib/ansible
  executable location = /root/tmp/ansible/bin/ansible
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]
Using /etc/ansible/ansible.cfg as config file
Parsed /root/windows/hosts inventory source with ini plugin
META: ran handlers
Using module file /root/tmp/ansible/lib/ansible/modules/windows/setup.ps1
<192.168.0.169> ESTABLISH WINRM CONNECTION FOR USER: Administrator on PORT 5986 TO 192.168.0.169
EXEC (via pipeline wrapper)
192.168.0.169 | FAILED! => {
    "changed": false,
    "msg": "Get-AnsibleParam: Parameter 'fact_path' has an invalid path 'C:\u0007' specified."
}
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/system/setup.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/setup.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
This isn't a problem with Ansible but just just generic quoting and argument escaping rules that you need to be aware of. We've created a guide to help people with this [here](http://docs.ansible.com/ansible/devel/user_guide/windows_usage.html#path-formatting-for-windows). Using the command line to specify paths are even more complex as it has a few different rules you need to be aware off. For this case I would just substitute `\` with `/` but this won't work in all cases, e.g.

```
ansible mygroup -m setup -a "fact_path='C:/a'"
```

When dealing with writting paths in yaml, you have a few options that you should try and do

```
path: C:\temp
path: 'C:\temp'
path: "C:\\temp"
```

I only needed to escape the last one as it was enclosed in double quotes, not quoting or single quotes work as is and usually my preferred way of writting paths.

**Comment by erik-saarts** [comment]
I see! I should have realized that myself, but it's too late now! As a newcomer to ansible I have only gone through minimal required amount of documentation for my needs and this guide obviously I did not read. I did read http://docs.ansible.com/ansible/latest/setup_module.html though, and last example there is very similar to what I had problem with (# ansible windows -m setup -a "fact_path='c:\custom_facts'"). It probably will not hurt to mention guide you provided there also.
Thank you for help!
